wordSearch.c

```diff
@@ -73,29 +88,36 @@ int main(void)
 
     fprintf(fp, "</p>");
     fprintf(fp, "</table></body></html>");
+
     fclose(fp);
+    fclose(wordFile);
 
     return 0;
 }
 
-void wordPick(FILE *fp, char board[DIM_X][DIM_Y])
+void loadWords(FILE *wordFile, char words[FILE_WORD_COUNT][MAX_WORD_LENGTH])
+{
+    int wordsRead = 0;
+    for (int i = 0; fgets(words[i], BUFSIZ, wordFile); i++) {
+        wordsRead++;
+    }
+    if (wordsRead != FILE_WORD_COUNT) {
+        fprintf(stderr, "Reading error. Expected %d words but got %d\n",
+            FILE_WORD_COUNT, wordsRead);
+        exit(1);
+    }
+}
+
+void wordPick(FILE *fp, char board[DIM_X][DIM_Y],
+              char words[FILE_WORD_COUNT][MAX_WORD_LENGTH])
 {
     int  x, y, length, type, direction, lineNum, count;
     char wordBuff[BUFSIZ] = "";
 
     for (int i = 0; i < NUM_WORDS; i++) {
         do {
-            lineNum  = (rand() % FILE_WORD_COUNT);
-            FILE *xd = fopen(WORD_INPUT, "r");
-            if (!xd) {
-                fprintf(stderr, "Could not open %s\n", WORD_INPUT);
-                perror("fopen");
-            }
-
-            for (int i = 0; i < lineNum; i++) {
-                fgets(wordBuff, BUFSIZ, xd);
-            }
-            fclose(xd);
+            lineNum = rand() % FILE_WORD_COUNT;
+            strcpy(wordBuff, words[lineNum - 1]);
```

Comment:
magnum opus - instead of manually reading in each word from `fgets()` manually, it now just pulls the respective word from the `words[][]` array. though for some reason in order to replicate the same results in debugging I had to subtract 1